ddl/placement/bundle.go

```diff
@@ -49,6 +50,200 @@ func NewBundle(id int64) *Bundle {
 	}
 }
 
+// NewBundleFromConstraintsOptions will transform constraints options into the bundle.
+func NewBundleFromConstraintsOptions(options *model.PlacementSettings) (*Bundle, error) {
+	if options == nil {
+		return nil, fmt.Errorf("%w: options can not be nil", ErrInvalidPlacementOptions)
+	}
+
+	if len(options.PrimaryRegion) > 0 || len(options.Regions) > 0 || len(options.Schedule) > 0 {
+		return nil, fmt.Errorf("%w: should be [LEADER/VOTER/LEARNER/FOLLOWER]_CONSTRAINTS=.. [VOTERS/FOLLOWERS/LEARNERS]=.., mixed other sugar options %s", ErrInvalidPlacementOptions, options)
+	}
+
+	constraints := options.Constraints
+	leaderConstraints := options.LeaderConstraints
+	learnerConstraints := options.LearnerConstraints
+	followerConstraints := options.FollowerConstraints
+	voterConstraints := options.VoterConstraints
+	followerCount := options.Followers
+	voterCount := options.Voters
+	learnerCount := options.Learners
+
+	if voterCount+followerCount == 0 {
+		return nil, fmt.Errorf("%w: not enough voter, please specify voter or follower count", ErrInvalidPlacementOptions)
+	}
+
+	CommonConstraints, err := NewConstraintsFromYaml([]byte(constraints))
```

Comment:
In create policy check, the xxx_constraint can be a dict or list, only common constraint is a list  >.<

similar to `NewMergeRules` , can we reuse that? anyway, i'm not strong on this